Frameworks/AddressBook/ABRecord.mm

```diff
@@ -80,18 +150,458 @@ ABRecordType ABRecordGetRecordType(ABRecordRef record) {
+
+        person.contact.middleName = middleName;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonPrefixProperty) {
+        NSString* prefix = (__bridge NSString*)value;
+        if (!_checkLength(prefix, kABPersonPrefixLength, error, @"prefix")) {
+            return false;
+        }
+
+        person.contact.honorificNamePrefix = prefix;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonSuffixProperty) {
+        NSString* suffix = (__bridge NSString*)value;
+        if (!_checkLength(suffix, kABPersonSuffixLength, error, @"suffix")) {
+            return false;
+        }
+
+        person.contact.honorificNameSuffix = suffix;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonNicknameProperty) {
+        NSString* nickname = (__bridge NSString*)value;
+        person.contact.nickname = nickname;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonFirstNamePhoneticProperty) {
+        NSString* phoneticFirstName = (__bridge NSString*)value;
+        if (!_checkLength(phoneticFirstName, kABPersonFirstNamePhoneticLength, error, @"first name phonetic")) {
+            return false;
+        }
+
+        person.contact.yomiGivenName = phoneticFirstName;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonLastNamePhoneticProperty) {
+        NSString* phoneticLastName = (__bridge NSString*)value;
+        if (!_checkLength(phoneticLastName, kABPersonLastNamePhoneticLength, error, @"last name phonetic")) {
+            return false;
+        }
+
+        person.contact.yomiFamilyName = phoneticLastName;
+        _updateManager(person);
+        return true;
+    }
+
+    // Cases for job-related properties.
+    if (contactProperty == kABPersonOrganizationProperty) {
+        NSString* organization = (__bridge NSString*)value;
+        if (!_checkLength(organization, kABPersonOrganizationLength, error, @"organization")) {
+            return false;
+        }
+
+        NSMutableArray* jobInfo = person.contact.jobInfo;
+        if ([jobInfo count] == 0) {
+            [jobInfo addObject:[WACContactJobInfo make]];
+        }
+
+        WACContactJobInfo* job = jobInfo[0];
+        job.companyName = organization;
+        _updateManager(person);
+        return true;
+    } else if (contactProperty == kABPersonJobTitleProperty) {
+        NSString* jobTitle = (__bridge NSString*)value;
+        if (!_checkLength(jobTitle, kABPersonJobTitleLength, error, @"job title")) {
+            return false;
+        }
+
+        NSMutableArray* jobInfo = person.contact.jobInfo;
+        if ([jobInfo count] == 0) {
+            [jobInfo addObject:[WACContactJobInfo make]];
+        }
+
+        WACContactJobInfo* job = jobInfo[0];
+        job.title = jobTitle;
+        _updateManager(person);
+        return true;
```

Comment:
Updated to avoid multiple return true's

---
In reply to: [76670284](https://github.com/Microsoft/WinObjC/pull/935#discussion_r76670284) [](ancestors = 76670284)